scripts/dev/dockerfile_generator.py

```diff
@@ -1,60 +1,99 @@
 import jinja2
 import argparse
+import os
+import sys
+from typing import List, Dict, Union
 
 
-def operator_params():
+def operator_params(files_to_add: List[str]) -> Dict[str, Union[bool, str, List[str]]]:
     return {
         "builder": True,
         "builder_image": "golang",
         "base_image": "registry.access.redhat.com/ubi8/ubi-minimal:latest",
+        "files_to_add": files_to_add,
     }
 
 
-def test_runner_params():
+def test_runner_params(
+    files_to_add: List[str],
+) -> Dict[str, Union[bool, str, List[str]]]:
     return {
         "builder": True,
         "builder_image": "golang",  # TODO: make this image smaller. There were errors using alpine
         "base_image": "registry.access.redhat.com/ubi8/ubi-minimal:latest",
+        "files_to_add": files_to_add,
     }
 
 
-def e2e_params():
+def e2e_params(files_to_add: List[str]) -> Dict[str, Union[bool, str, List[str]]]:
     return {
         "base_image": "golang",  # TODO: make this image smaller, error: 'exec: "gcc": executable file not found in $PATH' with golang:alpine
+        "files_to_add": files_to_add,
     }
 
 
-def unit_test_params():
+def unit_test_params(files_to_add: List[str]) -> Dict[str, Union[bool, str, List[str]]]:
     return {
         "base_image": "golang",
+        "files_to_add": files_to_add,
     }
 
 
-def render(image_name):
+def python_formatting_params(
+    files_to_add, script
+) -> Dict[str, Union[bool, str, List[str]]]:
+    return {
+        "base_image": "python:slim",
+        "files_to_add": files_to_add,
+        "script_location": script,
+    }
+
+
+def render(image_name, files_to_add, script_location) -> str:
```

Comment:
can we add type hinting to arguments too?

Reply:
> Dict[str, Union[bool, str, List[str]]] is even less readable than Dict I think! Was this a mypy error?

This was suggested by @LouisPlisso. I don't have a strong preference but I will get his suggestion of defining that one as a type and using that one each time.

> can we add type hinting to arguments too?


Sure thing thanks!